### PR TITLE
Feature/calendar

### PR DESCRIPTION
**Pull request by JoshEflin**
Added  a Picnic Class and several functions that transform data from the database into an object that google calendar can read. Completed OAuth flow so that a user is directed to google calendar in a separate tab. Data about which picnic the user wants to add is stored in a state in the query parameters which is passed through the oauth chain until it is finally passed into the googEventCreator.  Several irrelevant files were deleted  and commented out code was deleted as well. EXTRA- fixed a bug where a user can join the same event twice using a hand written filter method

### CODE REVIEW

File: controllers/htmlRoutes/htmlRoutes.js [hunk 1]
```diff
@@ -57,12 +57,20 @@ router.get('/my-picnics', withAuth, async (req, res) => {
...
             return picnic.get({ plain: true });
         });
+
+        const attending =[] 
+        for (let i = 0 ; i< attendingArr.length; i++){
+           // console.log(attendingArr[i].id)
+           // console.log(hosting[0].id)
+           if (attendingArr[i].id !== hosting[0].id)
+        attending.push(attendingArr[i])
+        }
```
Comment (JoshEflin) [inline controllers/htmlRoutes/htmlRoutes.js:73]: only pushes events that a user is not hosting to attending array.  as long as the user is only hosting one picnic( we can use a nested for loop to accomplish the same if hosting more than one picnic)

**Comment by miacias** [comment]
tested, notes for the group:
currently only capable of adding first event to calendar. need to adjust query selector and target to querySelectorAll and currentTarget

**Comment by miacias** [comment]
testing again
- note: user needs to go through multiple steps to "trust" the red-knights but it is possible to accept (google security unavoidable)
- calendar events now add to calendar based on event clicked

POTENTIAL UNWANTED BEHAVIOR??:

1. a user who adds an event to their personal calendar also adds invitees to the their personal calendar. they do not receive google emails or alerts, but note bullet point 2
2. a user who deletes then a calendar event from their personal calendar **also removes and automatically alerts "invitees"** (unless user checks alternate option)
3. potential solution: remove attendee email addresses. perhaps an alternative would be to list them by first name and last initial in the Summary if attendance list is desired

**Review by miacias — APPROVED** [review]
incredible work!

**Pull request merged**